lib/adagrams.rb

```diff
@@ -0,0 +1,101 @@
+require "csv"
+
+def draw_letters
+  pool = [
+    "A", "A", "A", "A", "A", "A", "A", "A", "A",
+    "B", "B",
+    "C", "C",
+    "D", "D", "D", "D",
+    "E", "E", "E", "E", "E", "E", "E", "E", "E", "E", "E", "E",
+    "F", "F",
+    "G", "G", "G",
+    "H", "H",
+    "I", "I", "I", "I", "I", "I", "I", "I", "I",
+    "J",
+    "K",
+    "L", "L", "L", "L",
+    "M", "M",
+    "N", "N", "N", "N", "N", "N",
+    "O", "O", "O", "O", "O", "O", "O", "O",
+    "P", "P",
+    "Q",
+    "R", "R", "R", "R", "R", "R",
+    "S", "S", "S", "S",
+    "T", "T", "T", "T", "T", "T",
+    "U", "U", "U", "U",
+    "V", "V",
+    "W", "W",
+    "X",
+    "Y", "Y",
+    "Z",
+  ]
+
+  hand = pool[1..98].sample(10)
+  return hand
+end
+
+def uses_available_letters?(input, letters_in_hand)
+  word = input.upcase
+  split_word = word.split("")
+
+  split_word.each do |letter|
+    index = letters_in_hand.index(letter)
+    if letters_in_hand.all? { letter } && index != nil
```

Comment:
What does the `letters_in_hand.all? { letter }` part do? This code actually just checks if `letter` is not `nil` or `false`! Sometimes, it's good to check that `letters_in_hand` has no `nil` or `false` elements, but in this case, I think it's okay to assume that won't be the case. In general, just checking `if index != nil` (aka, if `index` is not `nil`, it's because it found something in the last line) probably suffices